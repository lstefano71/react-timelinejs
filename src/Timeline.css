/* axis */

/*.axis { !* axis labels *!*/
    /*fill: #808080;*/
    /*font-family: sans-serif;*/
    /*font-size: 10px;*/
/*}*/

/*.axis line{ !* axis tick marks *!*/
    /*stroke-width : 1;*/
    /*stroke: grey;*/
    /*shape-rendering: crispEdges;*/
/*}*/

/*.axis path { !* axis line *!*/
    /*stroke-width : 1;*/
    /*stroke: grey;*/
    /*shape-rendering: crispEdges;*/
/*}*/

/* timeline band */

.timeline {
    position: relative;
}

.band { /* band background */
    fill: #FAFAFA;
}

/* labels */

.bandLabel {
    fill: #F0F0F0;
    font: 10px sans-serif;
    font-weight: bold;
}

.bandMinMaxLabel {
    fill: blue;
    font: 10px sans-serif;
    font-weight: bold;
}

.bandMidLabel {
    fill: red;
    font: 10px sans-serif;
    font-style: italic;
    font-weight: bold;
}

/* brush */

.brush .extent {
    stroke: gray;
    fill: blue;
    fill-opacity: .1;
}

.chart {
    fill: #EEEEEE;
}

.interval {
    fill: dodgerblue;
    stroke-width: 6;
    cursor : default;
    pointer-events: true;
}

.instant {
    fill: dodgerblue;
    stroke-width: 6;
    cursor : default;
    /*pointer-events: true;*/
}

.instantLabel {
    fill : blue;
    font: 10px sans-serif;
    shape-rendering: crispEdges;
}

.intervalLabel {
    fill : black;
    font: 10px sans-serif;
    shape-rendering: crispEdges;
}

.item {
    cursor : default;
    pointer-events: auto;
}

/*.svg {*/
    /*border-style: solid;*/
    /*border-width: 1px;*/
    /*border-color: black;*/
    /*background-color: #FFFFFF;*/
/*}*/

.tooltip {
    width: 0px;
    height: 0px;
    position: absolute;
    z-index: 2;
    visibility: hidden;
    color : black;
    cursor:default;
    pointer-events: none;
    display: flex;
}

.tooltip.top-left {

}
.tooltip.top-right {
    flex-direction: row-reverse;
}
.tooltip.bottom-right {
    flex-flow: row-reverse wrap-reverse;
}
.tooltip.bottom-left {
    flex-flow: column-reverse;
}

.tooltip-container {
    shape-rendering: crispEdges;
    padding: 4px;
    min-width: 300px;
    min-height: 100px;
    border: 1px solid;
    background-color: #FFFFEE;
}

.mouseover-line {
    stroke: #0984e3;
}

.mouseover-label {
    fill : white;
    font: 10px sans-serif;
    shape-rendering: crispEdges;
    text-anchor: middle;
}

.mouseover-label-background {
    fill: #0984e3;
}
