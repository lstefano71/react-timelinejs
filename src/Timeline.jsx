import React from 'react';
import timeline from './TimelineModel';
import './Timeline.css';
import equals from 'ramda/src/equals';
import PropTypes from 'prop-types';

const propTypes = {
    data: PropTypes.array.isRequired,
    lines: PropTypes.array,

    dataKey: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    trackHeight: PropTypes.number,
    label: PropTypes.bool,
    tooltips: PropTypes.bool,
    tooltipContent: PropTypes.func,
    brush: PropTypes.bool,
    brushRange: PropTypes.array,
    range: PropTypes.array,
    onBrush: PropTypes.func,
    onMouseover: PropTypes.func,
    onClick: PropTypes.func,
};

//
// const defaultProps = {
//     styles: {
//         label: {
//             fontFamily: 'Comic Sans MS',
//             color: 'green'
//         },
//         input: {
//             background: '#ddd',
//             border: '1px solid red'
//         }
//     }
// };

class Timeline extends React.Component {

    componentDidMount() {
        this.key = Math.round(Math.random() * 100000).toFixed(0) + '-' + new Date().getTime();
        this.createTimeline();
    }


    componentDidUpdate() {
    }

    componentWillReceiveProps(nextProps, nextState) {
        if (!equals(nextProps.data, this.props.data)
                || !equals(nextProps.lines, this.props.lines)) {
            this.timelineFn
                .data(nextProps.data)
                .lines(nextProps.lines)
                .update()
                .updateRange(nextProps.range)
                .updateBrushRange(nextProps.brushRange)
                .redraw();
        }
        if (!equals(nextProps.range, this.props.range)){
            this.timelineFn.updateRange(nextProps.range);
        }
        if (!equals(nextProps.brushRange, this.props.brushRange)){
            this.timelineFn.updateBrushRange(nextProps.brushRange);
        }
    }

    createTimeline = () => {

        const {data, dataKey, lines, height, width, trackHeight, label, tooltips, tooltipContent, brush, brushRange, onBrush, onMouseover, onClick, range} = this.props;
        const config = {
            dataKey,
            width,
            height,
            trackHeight,
            label,
            tooltips,
            tooltipContent,
            brush,
            brushRange,
            range,
            onBrush,
            onMouseover,
            onClick
        };
        this.timelineFn = timeline(this.div, config).create(data, lines);
        this.timelineFn.redraw();
    };

    render() {
        return <div ref={input => this.div = input}/>
    }
}

Timeline.propTypes = propTypes;
// Timeline.defaultProps = defaultProps;

export default Timeline;
