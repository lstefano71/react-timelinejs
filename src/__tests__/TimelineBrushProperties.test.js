import React from 'react';
import Timeline from '../Timeline';
import {mockData, mockLines, mockRange} from './Timeline.tester';

class StatefulTimeline extends React.Component {

    constructor(props){
        super(props);
        const data = mockData({size: 3});
        this.state = {
            data: data,
            lines: mockLines({}),
            range: mockRange(data)
        }
    }

    updateMockData = () => {
        const data = mockData({size: 4});
        this.setState({
            data: data,
            lines: mockLines({}),
            range: mockRange(data)
        });
    };

    render() {

        const {data, lines, range} = this.state;
        return (
            <div>
                <button id='update-mock-data' onClick={this.updateMockData}>generate</button>
                <Timeline data={data} dataKey='key' lines={lines} brush={true} brushRange={range}/>
            </div>
        )
    }


}


test('render update brush properties', () => {
    const wrapper = mount(<StatefulTimeline/>);
    expect(html(wrapper)).toMatchSnapshot();
    wrapper.find('#update-mock-data').simulate('click');
    expect(html(wrapper)).toMatchSnapshot();
});




