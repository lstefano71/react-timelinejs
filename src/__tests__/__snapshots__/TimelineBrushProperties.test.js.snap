// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render update brush properties 1`] = `
"
<div>
  <button id=\\"update-mock-data\\">
    generate
  </button>
  <div>
    <svg class=\\"svg\\"
         id=\\"svg-1\\"
         width=\\"700\\"
         height=\\"400\\"
    >
      <g transform=\\"translate(0,0)\\">
        <clipPath id=\\"chart-area\\">
          <rect width=\\"700\\"
                height=\\"400\\"
          >
          </rect>
        </clipPath>
        <g class=\\"chart\\"
           clip-path=\\"url(#chart-area)\\"
        >
          <g id=\\"band0\\"
             transform=\\"translate(0,0)\\"
          >
            <rect class=\\"band\\"
                  width=\\"700\\"
                  height=\\"380\\"
            >
            </rect>
            <path class=\\"line test\\"
                  d=\\"M-878.6692595940302 0 L-878.6692595940302 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-814.9444020848346 0 L-814.9444020848346 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-751.2195445756389 0 L-751.2195445756389 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-687.4946870664434 0 L-687.4946870664434 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-623.7698295572478 0 L-623.7698295572478 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-560.0449720480522 0 L-560.0449720480522 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-496.3201145388566 0 L-496.3201145388566 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-432.59525702966096 0 L-432.59525702966096 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-368.87039952046536 0 L-368.87039952046536 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-305.1455420112698 0 L-305.1455420112698 380\\"
            >
            </path>
            <path d=\\"M0 0 L0 380\\"
                  class=\\"mouseover-line\\"
                  style=\\"display: none;\\"
            >
            </path>
            <rect height=\\"16\\"
                  width=\\"75\\"
                  y=\\"364\\"
                  class=\\"mouseover-label-background\\"
                  style=\\"display: none;\\"
            >
            </rect>
            <text x=\\"0\\"
                  y=\\"376\\"
                  class=\\"mouseover-label\\"
                  style=\\"display: none;\\"
            >
            </text>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"-1000\\"
                 width=\\"0\\"
                 style=\\"display: none;\\"
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 2
              </text>
            </svg>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"-1000\\"
                 width=\\"0\\"
                 style=\\"display: none;\\"
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 0
              </text>
            </svg>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"-1000\\"
                 width=\\"0\\"
                 style=\\"display: none;\\"
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 1
              </text>
            </svg>
          </g>
          <g class=\\"axis\\"
             transform=\\"translate(0,380)\\"
             fill=\\"none\\"
             font-size=\\"10\\"
             font-family=\\"sans-serif\\"
             text-anchor=\\"middle\\"
          >
            <path class=\\"domain\\"
                  stroke=\\"#000\\"
                  d=\\"M0.5,6V0.5H700.5V6\\"
            >
            </path>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(13.978745534708198,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                02 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(77.7036030439038,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                03 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(141.4284605530994,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                04 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(205.15331806229497,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                05 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(268.8781755714906,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                06 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(332.6030330806862,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                07 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(396.32789058988175,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                08 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(460.0527480990774,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                09 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(523.777605608273,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                10 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(587.5024631174686,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                11 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(651.2273206266642,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                Jul 15
              </text>
            </g>
          </g>
        </g>
      </g>
    </svg>
    <div class=\\"tooltip\\"
         id=\\"tooltip-1\\"
         style=\\"visibility: hidden;\\"
    >
    </div>
  </div>
</div>
"
`;

exports[`render update brush properties 2`] = `
"
<div>
  <button id=\\"update-mock-data\\">
    generate
  </button>
  <div>
    <svg class=\\"svg\\"
         id=\\"svg-1\\"
         width=\\"700\\"
         height=\\"400\\"
    >
      <g transform=\\"translate(0,0)\\">
        <clipPath id=\\"chart-area\\">
          <rect width=\\"700\\"
                height=\\"400\\"
          >
          </rect>
        </clipPath>
        <g class=\\"chart\\"
           clip-path=\\"url(#chart-area)\\"
        >
          <g id=\\"band0\\"
             transform=\\"translate(0,0)\\"
          >
            <rect class=\\"band\\"
                  width=\\"700\\"
                  height=\\"380\\"
            >
            </rect>
            <path class=\\"line test\\"
                  d=\\"M-406.7754357250873 0 L-406.7754357250873 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-340.03620338516214 0 L-340.03620338516214 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-273.29697104523706 0 L-273.29697104523706 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-206.55773870531186 0 L-206.55773870531186 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-139.8185063653867 0 L-139.8185063653867 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-73.07927402546154 0 L-73.07927402546154 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M-6.34004168553639 0 L-6.34004168553639 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M60.39919065438876 0 L60.39919065438876 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M127.13842299431393 0 L127.13842299431393 380\\"
            >
            </path>
            <path class=\\"line test\\"
                  d=\\"M193.87765533423908 0 L193.87765533423908 380\\"
            >
            </path>
            <path d=\\"M0 0 L0 380\\"
                  class=\\"mouseover-line\\"
                  style=\\"display: none;\\"
            >
            </path>
            <rect height=\\"16\\"
                  width=\\"75\\"
                  y=\\"364\\"
                  class=\\"mouseover-label-background\\"
                  style=\\"display: none;\\"
            >
            </rect>
            <text x=\\"0\\"
                  y=\\"376\\"
                  class=\\"mouseover-label\\"
                  style=\\"display: none;\\"
            >
            </text>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"-1000\\"
                 width=\\"0\\"
                 style=\\"display: none;\\"
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 2
              </text>
            </svg>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"-1000\\"
                 width=\\"0\\"
                 style=\\"display: none;\\"
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 3
              </text>
            </svg>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"672.0191133745951\\"
                 width=\\"30.032654552966278\\"
                 style
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 0
              </text>
            </svg>
            <svg y=\\"4\\"
                 height=\\"19\\"
                 class=\\"part interval\\"
                 x=\\"-1000\\"
                 width=\\"0\\"
                 style=\\"display: none;\\"
            >
              <rect width=\\"100%\\"
                    height=\\"100%\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"10%\\"
                    y=\\"0\\"
                    width=\\"20%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"40%\\"
                    y=\\"0\\"
                    width=\\"9.999999999999998%\\"
                    height=\\"19\\"
              >
              </rect>
              <rect class=\\"step custom_1\\"
                    x=\\"55.00000000000001%\\"
                    y=\\"0\\"
                    width=\\"19.999999999999996%\\"
                    height=\\"19\\"
              >
              </rect>
              <text class=\\"intervalLabel\\"
                    x=\\"1\\"
                    y=\\"10\\"
              >
                event 1
              </text>
            </svg>
          </g>
          <g class=\\"axis\\"
             transform=\\"translate(0,380)\\"
             fill=\\"none\\"
             font-size=\\"10\\"
             font-family=\\"sans-serif\\"
             text-anchor=\\"middle\\"
          >
            <path class=\\"domain\\"
                  stroke=\\"#000\\"
                  d=\\"M0.5,6V0.5H700.5V6\\"
            >
            </path>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(60.89919065438876,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                07 AM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(127.63842299431393,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                08 AM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(194.37765533423908,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                09 AM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(261.1168876741642,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                10 AM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(327.85612001408936,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                11 AM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(394.5953523540146,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                12 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(461.3345846939397,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                01 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(528.0738170338649,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                02 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(594.81304937379,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                03 PM
              </text>
            </g>
            <g class=\\"tick\\"
               opacity=\\"1\\"
               transform=\\"translate(661.5522817137152,0)\\"
            >
              <line stroke=\\"#000\\"
                    y2=\\"6\\"
              >
              </line>
              <text fill=\\"#000\\"
                    y=\\"9\\"
                    dy=\\"0.71em\\"
              >
                04 PM
              </text>
            </g>
          </g>
        </g>
      </g>
    </svg>
    <div class=\\"tooltip\\"
         id=\\"tooltip-1\\"
         style=\\"visibility: hidden;\\"
    >
    </div>
  </div>
</div>
"
`;
